contracts/BC_fusion/StakePool.sol

```diff
@@ -0,0 +1,266 @@
+
+        emit UnbondRequested(_delegator, _sharesAmount, _bnbAmount, request.unlockTime);
+        return _bnbAmount;
+    }
+
+    function redelegate(address _delegator, uint256 _sharesAmount) external onlyStakeHub returns (uint256) {
+        require(_sharesAmount <= balanceOf(_delegator), "INSUFFICIENT_BALANCE");
+
+        // calculate the BNB amount and update state
+        uint256 _bnbAmount = getPooledBNBByShares(_sharesAmount);
+        _burn(_delegator, _sharesAmount);
+        _totalPooledBNB -= _bnbAmount;
+
+        (bool success,) = STAKE_HUB_ADDR.call{value: _bnbAmount}("");
+        require(success, "TRANSFER_FAILED");
+        return _bnbAmount;
+    }
+
+    function claim(address payable _delegator, uint256 number) external onlyStakeHub nonReentrant returns (uint256) {
+        if (_delegator == validator) {
+            require(!_freeze, "FROZEN");
+        }
+
+        require(_unbondRequestsQueue[_delegator].length() != 0, "NO_UNBOND_REQUEST");
+        // number == 0 means claim all
+        if (number == 0) {
+            number = _unbondRequestsQueue[_delegator].length();
+        }
+        if (number > _unbondRequestsQueue[_delegator].length()) {
+            number = _unbondRequestsQueue[_delegator].length();
+        }
+        require(number <= MAX_CLAIM_NUMBER, "TOO_MANY_REQUESTS"); // prevent too many loop in one transaction
+
+        uint256 _totalShares;
+        uint256 _totalBnbAmount;
+        while (number != 0) {
+            bytes32 hash = _unbondRequestsQueue[_delegator].front();
+            UnbondRequest memory request = _unbondRequests[hash];
+            if (block.timestamp < request.unlockTime) {
+                break;
+            }
+
+            _totalShares += request.sharesAmount;
+            _totalBnbAmount += request.bnbAmount;
+
+            // remove from the queue
+            _unbondRequestsQueue[_delegator].popFront();
+            delete _unbondRequests[hash];
+
+            number -= 1;
+        }
+
+        _lockedShares[_delegator] -= _totalShares;
+        uint256 gasLimit = IStakeHub(STAKE_HUB_ADDR).transferGasLimit();
+        (bool success,) = _delegator.call{value: _totalBnbAmount, gas: gasLimit}("");
+        require(success, "CLAIM_FAILED");
+
+        emit UnbondClaimed(_delegator, _totalShares, _totalBnbAmount);
+        return _totalBnbAmount;
+    }
+
+    function distributeReward() external payable onlyStakeHub {
+        uint256 _bnbAmount = msg.value;
+        _totalPooledBNB += _bnbAmount;
+        emit RewardReceived(_bnbAmount);
+    }
+
+    function slash(uint256 _slashBnbAmount) external onlyStakeHub returns (uint256) {
+        uint256 _securityDeposit = balanceOf(STAKE_HUB_ADDR);
+        uint256 _slashShares = getSharesByPooledBNB(_slashBnbAmount);
+
+        uint256 _remain;
+        if (_slashShares <= _securityDeposit) {
+            _totalPooledBNB -= _slashBnbAmount;
+            _burn(STAKE_HUB_ADDR, _slashShares);
```

Comment:
I think we should not burn because this will cause that other delegator of this validator will get more reward, this is agains to our design, what about transfer the stBNB to `systemreward` contract?